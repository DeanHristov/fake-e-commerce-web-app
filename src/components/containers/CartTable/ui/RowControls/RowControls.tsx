import { FC } from 'react';

export interface IControlsProps {
  onRemove: () => void;
  onWishList: () => void;
  className?: string;
}

const RowControls: FC<IControlsProps> = ({
  onRemove,
  onWishList,
  className = '',
}) => {
  return (
    <div className={`text-black text-sm ${className}`}>
      <span
        data-testid={'btn-remove'}
        onClick={onRemove}
        className="hover:underline hover:cursor-pointer hover:text-dark"
      >
        Remove
      </span>
      {' / '}
      <span
        data-testid={'btn-wish-list'}
        onClick={onWishList}
        className="hover:underline hover:cursor-pointer hover:text-dark"
      >
        Wish list
      </span>
    </div>
  );
};

export default RowControls;
