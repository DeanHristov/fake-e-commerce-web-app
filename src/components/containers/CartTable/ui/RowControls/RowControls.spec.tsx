import { render } from '@testing-library/react';

import RowControls from './RowControls';
import userEvent from '@testing-library/user-event';

describe('UI/Controls <RowControls {...} />', () => {
  it('Should render a default state of the Component', () => {
    const { container } = render(
      <RowControls onRemove={jest.fn} onWishList={jest.fn} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('Should be able to trigger "on-remove" action', async () => {
    const mockCallback = jest.fn();

    const { getByTestId } = render(
      <RowControls onRemove={mockCallback} onWishList={jest.fn} />,
    );

    await userEvent.click(getByTestId('btn-remove'));
    expect(mockCallback).toHaveBeenCalled();
  });

  it('Should be able to trigger "on-wishList" action', async () => {
    const mockCallback = jest.fn();

    const { getByTestId } = render(
      <RowControls onRemove={jest.fn} onWishList={mockCallback} />,
    );

    await userEvent.click(getByTestId('btn-wish-list'));

    expect(mockCallback).toHaveBeenCalled();
  });
});
