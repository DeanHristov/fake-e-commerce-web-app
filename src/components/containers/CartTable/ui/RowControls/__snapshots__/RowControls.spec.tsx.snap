// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Controls <RowControls {...} /> Should render a default state of the Component 1`] = `
<div>
  <div
    class="text-black text-sm "
  >
    <span
      class="hover:underline hover:cursor-pointer hover:text-dark"
      data-testid="btn-remove"
    >
      Remove
    </span>
     / 
    <span
      class="hover:underline hover:cursor-pointer hover:text-dark"
      data-testid="btn-wish-list"
    >
      Wish list
    </span>
  </div>
</div>
`;
